include/flyft/grid_interpolator.h

```diff
@@ -15,13 +16,17 @@ class GridInterpolator
         ~GridInterpolator();
         
         double operator()(double x, double y, double z) const;   
+        std::tuple<double, double , double> getBounds() const;
         
     private:
         ThreeDimensionalIndex n_;
         double* data_;
         double dx_;
         double dy_;
         double dz_;
+        double x_;
+        double cutoff_;
+        double rho_;
```

Comment:
Why are these variables here? They seem specific to your application, not to the general interpolation problem. I think you want to declare a bounds variable instead:
```suggestion
        std::tuple<double, double, double> bounds_;
```
then compute it when you read the file.